### PR TITLE
application logging

### LINKED ISSUE #19: Logging
- Application logging setup using logger and enhanced logging with rich.
- Documentation with examples of log entries and their significance.

### PR DESCRIPTION
**Pull request by allanzguan**
adding application logging

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -178,6 +178,11 @@ cython_debug/
 
 # hydra directory
 outputs
+<<<<<<< 19-logging
```
Comment (HenrydeB) [inline .gitignore]: I think this is left over from a merge conflict, could you clear up the merge conflict annotations here, on line 184, and on line 188?

**Review by HenrydeB — CHANGES_REQUESTED** [review]
Looks fine outside of the merge conflict annotations on the .gitignore. 

The ReadME also needs to be updated with the info you did on this task, were you going to add that in another branch? I'm wondering because I'm probably going to do it for every branch I work on (otherwise I would forget). Maybe I'll just do a ReadME branch, I haven't decided yet

File: .gitignore (hunk 1 above)
Comment (allanzguan) [reply .gitignore]: resolved

**Comment by allanzguan** [comment]
@HenrydeB Thanks for reminding, I have updated the README with the application logging stuff

**Review by HenrydeB — APPROVED** [review]
Update looks good

**Pull request merged**